src/libs/ReportUtils.js

```diff
@@ -1148,15 +1146,7 @@ function isWaitingForIOUActionFromCurrentUser(report, allReportsDict = null) {
         return true;
     }
 
-    let reportToLook = report;
-    if (report.iouReportID) {
-        const iouReport = allAvailableReports[`${ONYXKEYS.COLLECTION.REPORT}${report.iouReportID}`];
-        if (iouReport) {
-            reportToLook = iouReport;
-        }
-    }
-    // Money request waiting for current user to Pay (from chat or from iou report)
-    if (reportToLook.ownerAccountID && (reportToLook.ownerAccountID !== currentUserAccountID || currentUserAccountID === reportToLook.managerID) && reportToLook.hasOutstandingIOU) {
+    if (report.hasOutstandingIOU && report.ownerAccountID && (report.ownerAccountID !== currentUserAccountID || currentUserAccountID === report.managerID)) {
```

Comment:
It would be great to add a comment here to explain what exactly we are checking for here in plain english, or in other words list out the scenarios for when the `isWaitingForIOUActionFromCurrentUser` should return true so contributors can also understand this faster and easier

Reply:
Are you suggesting to restore the comment: "// Money request waiting for current user to Pay (from chat or from iou report)" ? That makes sense to me

Reply:
reading you comment again, were you suggesting to add the list of cases this function returns `true` in the documentation of the function?

Reply:
Yeah haha, basically list out in plain English when the green dot should show